Wyam.Core/Modules/Sitemap.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Wyam.Common.Documents;
+using Wyam.Common.IO;
+using Wyam.Common.Modules;
+using Wyam.Common.Pipelines;
+using Wyam.Core.Documents;
+
+namespace Wyam.Core.Modules
+{
+    public class Sitemap : IModule
+    {
+        private string sitemapFilename;
+        private Func<string, string> locationFormatter;
+        private static readonly string[] ChangeFrequencies = new string[] { "always", "hourly", "daily", "weekly", "monthly", "yearly", "never" };
+
+        public Sitemap(string sitemapFilename, Func<string, string> locationFormatter = null)
+        {
+            this.sitemapFilename = sitemapFilename;
+            this.locationFormatter = locationFormatter;
+        }
+
+        public IEnumerable<IDocument> Execute(IReadOnlyList<IDocument> inputs, IExecutionContext context)
+        {
+            var outputFilename = Path.Combine(context.OutputFolder, this.sitemapFilename);
+            var docs = context.Documents.Where(f => f.Metadata.ContainsKey(MetadataKeys.SitemapItem)).ToArray();
```

Comment:
Can we use `inputs` here instead of getting all documents from the context? That'll also require a `Documents()` module in the example pipeline to feed the site map.

This will allow fine-grained control of which documents the site map covers. It'll also work better for scenarios with "throw away" documents (for example, in the new Wyam web site I'll be launching soon, one pipeline finishes with source code documents that are only meant to drive other pipelines, not be output as pages).